### PR TITLE
dsmr add tariff

### LINKED ISSUE #2547: DSMR: sensor for actual tariff is missing
### The problem

The DSMR telegram contains a line with the actual tariff.
But the sensor for the actual tariff is missing.

### Which version of ESPHome has the issue?

2021.9.3

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2021.10.4

### What platform are you using?

ESP8266

### Board

slimmelezer

### Component causing the issue

dsmr

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

In the DSMR 2 telegram the line with the actual tariff is:
0-0:96.14.0(0002)
0001 = off-peak
0002 = peak

### PR DESCRIPTION
**Pull request by StijnVdd**
## Description:

**Related issue (if applicable):** fixes https://github.com/esphome/issues/issues/2547 

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#2745
## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Review by mmakaay — CHANGES_REQUESTED** [review]
The PR is done against the `current` branch.
I think the appropriate route would be to create it against `next` instead.
From there, the doc change will to to `beta` and will eventually end up in `current`.
(@oxan correct me if I'm wrong)

**Comment by oxan** [comment]
> The PR is done against the `current` branch. I think the appropriate route would be to create it against `next` instead. From there, the doc change will to to `beta` and will eventually end up in `current`. (@oxan correct me if I'm wrong)

Usually, for things corresponding to a PR, this is correct. However, given [my comment](https://github.com/esphome/esphome/pull/2754#issuecomment-973004839) in the PR, I think for once it's appropriate to target current here.

**Comment by StijnVdd** [comment]
As mentioned, it turns out only the docs are missing. 
@oxan Should/can I add the other text sensors too in this PR?

**Comment by oxan** [comment]
>  @oxan Should/can I add the other text sensors too in this PR?

Yes, definitely, that'd be great.

File: components/sensor/dsmr.rst [hunk 1]
```diff
@@ -224,12 +224,55 @@ Configuration variables:
...
+
+- **water_equipment_id** (*Optional*): Water Equipment ID
+
+   - **name** (**Required**, string): The name for the water_equipment_id sensor.
+   - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+   - All other options from :ref:`Text Sensor <config-text_sensor>`.
+
+- **sub_equipment_id** (*Optional*): Sub Equipment ID
+
+   - **name** (**Required**, string): he name for the sub_equipment_id sensor.
```
Comment (oxan) [inline components/sensor/dsmr.rst]:
```suggestion
   - **name** (**Required**, string): The name for the sub_equipment_id sensor.
```

**Review by oxan — APPROVED** [review]
Thanks!

**Comment by mmakaay** [comment]
Thanks @StijnVdd !

**Pull request merged**